### DISCUSSION TITLE
Follow `latest` docker tag

### DISCUSSION THREAD

**Discussion by DerRockWolf** [renovatebot/renovate#14015] [Request Help] [answered]
The official docker images often have unstable versions that are being tagged like the stable images.

An example: the current stable [busybox](https://hub.docker.com/_/busybox) image version/tag is `1.34.1`. This image is also tagged with `latest`. The repository also contains an unstable image tagged with `1.35.0`.

If I use `busybox:1.34.1` in a k8s manifest renovate will create a PR that bumps the tag to `1.35.0` (the unstable version).

Is there a way to follow along with the `latest` tag for those cases or are there other methods to prevent renovate from creating PRs for versions that aren't tagged with `latest`

**Comment by viceice** [comment, marked as answer]
No default way for renovate, it's because the docker API don't provide this information out of the box.

We would need to make one request per tag to get that information and we don't like to do that.

**Reply by rarkins** [reply]
Correct. Unlike npm, where their package manifest indicates which version is latest, Docker doesn't do that. You would be better to use the `latest` tag and let Renovate update sha256 values. And this image ideally shouldn't publish unstable releases using stable semver values like 1.35.0

**Reply by DerRockWolf** [reply]
Thank you for your answers. 

> You would be better to use the latest tag and let Renovate update sha256 values.

That could work out for me but would throw another question.
Am I correct in assuming that I need to manually look up the digest and see what tags it has so that I know what version the digest refers to in order to know if there was a major version change and possibly breaking changes? (Sorry for the messy sentence)
Or are the version tag changes still visible in the PR?

> And this image ideally shouldn't publish unstable releases using stable semver values like 1.35.0

I totally agree with you, but they are just using the original version from the original applications, so there sadly isn't much I can do about it.

**Reply by viceice** [reply]
you can use `ubuntu:20.04@sha256:XXXX` . For some managers renovate supports diggest pinning, so renovate will add the digest for you

**Reply by DerRockWolf** [reply]
That works but if I would want to use `latest` to always refer to the stable version I have no idea what version the digest is referring to without manually checking, am I correct about that?

**Reply by rarkins** [reply]
You can't have everything all at once:
- Use latest tag
- Know which version that "actually" means
- Pin to sha256

Options 1+2 are mutually exclusive
